### PR TITLE
Add format

### PR DESCRIPTION
**Pull request by sol**

### CODE REVIEW

File: src/System/Logging/LogSink.hs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import           System.Logging.LogSink.Format
+
+data LogTarget = StdErr | SysLog
+  deriving (Eq, Show)
+
+data Sink = Sink {
+  sinkLevel :: Maybe LogLevel
+, sinkFormat :: Maybe String
+, sinkTarget :: LogTarget
+} deriving (Eq, Show)
```
Comment (soenkehahn) [inline src/System/Logging/LogSink.hs]: @sol: Why have you abandoned the idea of having `type Sink = Format -> LogRecord -> IO ()`? I would have implemented `LogLevel` filtering and formatting on top of that.

Comment (soenkehahn) [inline src/System/Logging/LogSink.hs]: I would also hope that the code would be much easier to comprehend then.

Comment (sol) [inline src/System/Logging/LogSink.hs]: @soenkehahn We now have that type in `System.Logging.LogSink.Core`.  We just don't have a type synonym for it.

Comment (sol) [inline src/System/Logging/LogSink.hs]: E.g. see the type of `stdErrSink :: Format -> LogSink` (note that `LogSink` has type `LogRecord -> IO`).

**Pull request merged**